### PR TITLE
Documentation for System.Net.Http.SocketsHttpHandler.ConnectCallback

### PR DESCRIPTION
**Pull request by aik-jahoda**
## Summary

Documentation of System.Net.Http.SocketsHttpHandler.ConnectCallback introduced in dotnet/runtime#41955

### CODE REVIEW

**Comment by aik-jahoda** [comment]
@geoffkizer, can you please review/edit the doc? I'm sure you will find more accurate way how to describe this API. I will appreciate any suggestion.

**Comment by geoffkizer** [comment]
There are xml doc comments for this stuff -- shouldn't they be getting picked up?

**Comment by aik-jahoda** [comment]
As far as I know they should be picked the first time the xml stub for the new API is created.

I updated it according the comments from the source.

**Comment by gewarren** [comment]
> As far as I know they should be picked the first time the xml stub for the new API is created.

That's not quite true, we don't bring in the comments when we create the stubs. But @carlossanlop has a DocsPortingTool you can use to import the doc comments after the stubs are in place. If we imported doc comments with the stubs, then any existing comments in the docs would be overwritten across all the docs.

File: xml/System.Net.Http/SocketsHttpConnectionContext.xml [hunk 1]
```diff
@@ -34,7 +34,7 @@
         <ReturnType>System.Net.DnsEndPoint</ReturnType>
       </ReturnValue>
       <Docs>
-        <summary>To be added.</summary>
+        <summary>The DnsEndPoint to be used by the ConnectCallback to establish the connection.</summary>
```
Comment (gewarren) [inline xml/System.Net.Http/SocketsHttpConnectionContext.xml:37]:
```suggestion
        <summary>Gets the DNS endpoint to be used by the <see cref="P:System.Net.Http.SocketsHttpHandler.ConnectCallback" /> to establish the connection.</summary>
```

File: xml/System.Net.Http/SocketsHttpConnectionContext.xml [hunk 2]
```diff
@@ -55,7 +55,7 @@
         <ReturnType>System.Net.Http.HttpRequestMessage</ReturnType>
       </ReturnValue>
       <Docs>
-        <summary>To be added.</summary>
+        <summary>The initial HttpRequestMessage that is causing the connection to be created.</summary>
```
Comment (gewarren) [inline xml/System.Net.Http/SocketsHttpConnectionContext.xml:58]:
```suggestion
        <summary>Gets the initial request message that is causing the connection to be created.</summary>
```

**Review by gewarren — APPROVED** [review]

**Comment by geoffkizer** [comment]
Proposed changes from @gewarren look good to me.

**Comment by gewarren** [comment]
Oops.

**Pull request merged**